### PR TITLE
feat: CloudFormation lambda provider

### PR DESCRIPTION
**Pull request by johnjcsmith**
Replaces the existing `LambdaProvider` which provisions Lambda functions manually using the AWS API with `LambdaCfnProvider` which uses CloudFormation to handle provisioning resources.

- CloudFormation template is stored in S3
- The `releaseDeployment` endpoint will **not** automatically mark a deployment as "active" instead, a schedule task finds all the "uploading"  status deployments and checks their status in CloudFormation
- The CLI already polls the `getDeployment` endpoint to wait for the `ready` status.

### CODE REVIEW

File: control-plane/src/modules/deployment/lambda-cfn-provider.ts [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+          FunctionName: functionName,
+        }),
+      );
+    } catch (error: any) {
+      console.error("Failed to trigger lambda", functionName, error);
+    }
+  }
+
+  // Scheduled job which checks for pending deployments and updates their status based on the result of the CloudFormation stack
+  public async startPollingDeployments() {
```
Comment (johnjcsmith) [inline control-plane/src/modules/deployment/lambda-cfn-provider.ts:115]: Temporary, I am looking into replacing this with an endpoint listening to SNS

**Review by nadeesha — APPROVED** [review]
🙌🏽

File: infrastructure/deployment/lambda-cfn.yaml [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+Resources:
+  DifferentialService:
+    Type: AWS::Lambda::Function
+    Properties:
+      FunctionName: !Ref FunctionName
+      Handler: !Ref Handler
+      Role:
+        'Fn::If':
+          - IsExecutionRoleEmpty
+          - !Sub 'arn:aws:iam::${AWS::AccountId}:role/DeploymentLambdaRuntimeRole'
```
Comment (tomellis91) [inline infrastructure/deployment/lambda-cfn.yaml:74]: Is this a role that was already created in AWS? 

Would it be worth creating a role per lambda in this template and associating that with the function?

Comment (johnjcsmith) [reply infrastructure/deployment/lambda-cfn.yaml:74]: This is created as part of [bootstrapping the environment](https://github.com/differentialhq/differential/blob/main/infrastructure/cfn.yaml#L52).

The `control-plane` can override these via [parameters](https://github.com/differentialhq/differential/pull/165/files#diff-dba2bfe46dd179980d16b3d3af6a03b096c47a826cda311d23fa9786eab59c44R160).

But yeah, we could easily enough create a new role for each deployment.

**Pull request merged**